### PR TITLE
[JBPM-9648] Improve some flaky tests around Spring boot projects in droolsjbpm-integration repo

### PR DESCRIPTION
**Pull request by afalhambra**
**[JBPM-9648](https://issues.redhat.com/browse/JBPM-9648)**: Improve some flaky tests around Spring boot projects in droolsjbpm-integration repo

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java [hunk 1]
```diff
@@ -122,8 +126,12 @@ public void setup() {
     
     @After
     public void cleanup() {
-        kieServicesClient.disposeContainer(containerId);
-        kieServicesClient.disposeContainer(containerId2); 
+        if (kieServicesClient != null) {
+            ServiceResponse<Void> response = kieServicesClient.disposeContainer(containerId);
+            logger.info("Container {} disposed with response - {}", containerId, response.getMsg());
+            kieServicesClient.disposeContainer(containerId2);
+            logger.info("Container {} disposed with response - {}", containerId, response.getMsg());
```
Comment (gmunozfe) [inline kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java:133]: wrong trace, it should be bind to containerId2
```suggestion
            logger.debug("Container {} disposed with response - {}", containerId2, response.getMsg());
```

File: kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java [hunk 2]
```diff
@@ -122,8 +126,12 @@ public void setup() {
     
     @After
     public void cleanup() {
-        kieServicesClient.disposeContainer(containerId);
-        kieServicesClient.disposeContainer(containerId2); 
+        if (kieServicesClient != null) {
+            ServiceResponse<Void> response = kieServicesClient.disposeContainer(containerId);
+            logger.info("Container {} disposed with response - {}", containerId, response.getMsg());
```
Comment (gmunozfe) [inline kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java:131]: As it's a disposal info, probably debug level is enough, wdyt?

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Overall looks good, just a comment about a typo in a trace and also a suggestion regarding trace level

File: kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java (hunk 2 above)
Comment (afalhambra) [reply kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java:131]: I left it as `info` on purpose as some of these failed tests are having problems when disposing the container, and in the current logs this information is not shown, so it's impossible to figure out why this is happening.

File: kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java (hunk 1 above)
Comment (afalhambra) [reply kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java]: thanks a lot, I missed that!

File: kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java [hunk 3]
```diff
@@ -122,8 +126,12 @@ public void setup() {
...
+            kieServicesClient.disposeContainer(containerId2);
```
Comment (gmunozfe) [inline kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java:132]: Missing response assignation to be traced in the next line (if not, there's the response from the first disposal)
```suggestion
            response = kieServicesClient.disposeContainer(containerId2);
```

**Review by gmunozfe — CHANGES_REQUESTED** [review]

File: kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java (hunk 3 above)
Comment (afalhambra) [reply kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java]: Thanks! missed that. Done

**Review by gmunozfe — APPROVED** [review]
Looks good to me, stupendous work @afalhambra !

**Comment by afalhambra** [comment]
@mareknovotny could you please merge?

**Pull request merged**